arangod/RestServer/FlushFeature.cpp

```diff
@@ -48,43 +48,62 @@
 
 namespace arangodb {
 
-  // used by catch tests
-  #ifdef ARANGODB_USE_GOOGLE_TESTS
-    /*static*/ FlushFeature::DefaultFlushSubscription FlushFeature::_defaultFlushSubscription;
-  #endif
-
-  /// @brief base class for FlushSubscription implementations
-  class FlushFeature::FlushSubscriptionBase
-      : public FlushFeature::FlushSubscription {
-   public:
-    /// @brief earliest tick that can be released
-    virtual TRI_voc_tick_t tick() const = 0;
-
-   protected:
-    TRI_voc_tick_t const _databaseId;
-    arangodb::StorageEngine const& _engine;
-    TRI_voc_tick_t _tickCurrent; // last successful tick, should be replayed
-    TRI_voc_tick_t _tickPrevious; // previous successful tick, should be replayed
-    std::string const _type;
-
-    FlushSubscriptionBase(
-        std::string const& type, // subscription type
-        TRI_voc_tick_t databaseId, // vocbase id
-        arangodb::StorageEngine const& engine // vocbase engine
-    ): _databaseId(databaseId),
-       _engine(engine),
-       _tickCurrent(0), // default (smallest) tick for StorageEngine
-       _tickPrevious(0), // default (smallest) tick for StorageEngine
-       _type(type) {
+// used by catch tests
+#ifdef ARANGODB_USE_GOOGLE_TESTS
+  /*static*/ FlushFeature::DefaultFlushSubscription FlushFeature::_defaultFlushSubscription;
+#endif
+
+/// @brief base class for FlushSubscription implementations
+class FlushFeature::FlushSubscriptionBase
+    : public FlushFeature::FlushSubscription {
+ public:
+  virtual Result commit(VPackSlice data) override final {
+    if (data.isNone()) {
+      // upgrade tick without commiting actual marker
+      resetCurrentTick(_engine.currentTick());
+      return {};
     }
-  };
+
+    return commitImpl(data);
+  }
+
+  /// @brief earliest tick that can be released
+  virtual TRI_voc_tick_t tick() const = 0;
+
+ protected:
+  FlushSubscriptionBase(
+      std::string const& type, // subscription type
+      TRI_voc_tick_t databaseId, // vocbase id
+      arangodb::StorageEngine const& engine // vocbase engine
+  ): _databaseId(databaseId),
+     _engine(engine),
+     _tickCurrent(0), // default (smallest) tick for StorageEngine
+     _tickPrevious(0), // default (smallest) tick for StorageEngine
+     _type(type) {
+    _tickCurrent = _engine.currentTick();
+    resetCurrentTick(_tickCurrent);
```

Comment:
The second line sets `_tickPrevious = _tickCurrent`. Is that really intended? Looks to me we cannot be sure that these are already scanned.